components/server/src/common.rs

```diff
@@ -700,20 +700,23 @@ impl<T: fmt::Display + Send + 'static> Stop for LazyWorker<T> {
 }
 
 pub trait KvEngineBuilder: KvEngine {
-    fn build(disk_engine: RocksEngine) -> Self;
+    fn build(disk_engine: RocksEngine, pd_client: Option<Arc<RpcClient>>) -> Self;
```

Comment:
I'm thinking we can have two APIs, for is current build.
The other is build_hybrid, which has the extra parameter. 
And this parameter can be the abstraction of CacheHint
pub trait CacheHint {
    fn get_cache_hints()
    fn on_cache_hint_added()
    fn on_cache_hint_deleted()
}
Then we use pd client to implement it. The initialize of cache hint implementation can be in server.rs.

Reply:
Done. I did this in even a simpler way, using RangeHintService as a pure marker trait. I may later re-factor it and add members to the trait to simplify it.